firewood/src/db/proposal.rs

```diff
@@ -225,33 +225,23 @@ impl Proposal {
         }
 
         rev_inner.root_hash_staging.write(0, &hash.0)?;
-        let (root_hash_redo, root_hash_wal) = rev_inner.root_hash_staging.delta();
+        let (root_hash_redo, _) = rev_inner.root_hash_staging.delta();
 
         // schedule writes to the disk
-        rev_inner.disk_requester.write(
-            Box::new([
-                BufferWrite {
-                    store_id: store.merkle.payload.id(),
-                    delta: merkle_payload_redo,
-                },
-                BufferWrite {
-                    store_id: store.merkle.meta.id(),
-                    delta: merkle_meta_redo,
-                },
-                BufferWrite {
-                    store_id: rev_inner.root_hash_staging.id(),
-                    delta: root_hash_redo,
-                },
-            ]),
-            AshRecord(
-                [
-                    (MERKLE_META_STORE_ID, merkle_meta_wal),
-                    (MERKLE_PAYLOAD_STORE_ID, merkle_payload_wal),
-                    (ROOT_HASH_STORE_ID, root_hash_wal),
-                ]
-                .into(),
-            ),
-        );
+        rev_inner.disk_requester.write(Box::new([
+            BufferWrite {
+                store_id: store.merkle.payload.id(),
+                delta: merkle_payload_redo,
+            },
+            BufferWrite {
+                store_id: store.merkle.meta.id(),
+                delta: merkle_meta_redo,
+            },
+            BufferWrite {
+                store_id: rev_inner.root_hash_staging.id(),
+                delta: root_hash_redo,
+            },
+        ]));
```

Comment:
I'm not familiar with this part of the code -- what are these lines doing? It looks like they have something to do with "redo"s. Do we still need these if we don't need to construct prior revisions from disk? If so what are the things we're storing here used for?

Reply:
Here is needed as it is actually persisting the new changes from this revision.